internal/provider/models.go

```diff
@@ -12,3 +12,15 @@ type AblyApp struct {
 	Status    types.String `tfsdk:"status"`
 	TLSOnly   types.Bool   `tfsdk:"tls_only"`
 }
+
+// Ably Namespace
+type AblyNamespace struct {
```

Comment:
Some of the field types here have different names than the ones we have defined in "examples" (https://github.com/ably/terraform-provider-ably/blob/main/examples/resources/namespace.tf) when we were designing the HCL API - is this on purpose? 

If not - lets use the ones we have defined and reviewed before.

Reply:
Oh yeah, I meant to mention that but I forgot as I wrote this on Tuesday.

So in the link you said there's the line `app_id = ably_app.app1.app_id` however in reality the app resource has the filed `id` not `app_id`. Which I think is the better way anyway as it avoids saying app twice. So in a similar vein I gave the namespace an `id` field instead of `namespace_id`.

Reply:
So the problem is that in this case it is not actual ID, so to speak - it is 
> "The namespace or channel name that the channel rule will apply to."

I see that it is called "id" in the Control API, so I think we can keep the current version for now for the sake of consistency, but in general I don't think this name for the field is clear at all...

Reply:
Also, if we keep the new naming - we should update the example as part of this PR as well.

Reply:
Done.

Reply:
Also updated the ID description to be more relevant.

Reply:
Add follow up ticket #47